content/en/docs/v3.6/learning/api_guarantees.md

```diff
@@ -51,7 +51,7 @@ From the perspective of client, linearizability provides useful properties which
 
 For example, consider a client completing a write at time point 1 (*t1*). A client issuing a read at *t2* (for *t2* > *t1*) should receive a value at least as recent as the previous write, completed at *t1*. However, the read might actually complete only by *t3*. Linearizability guarantees the read returns the most current value. Without linearizability guarantee, the returned value, current at *t2* when the read began, might be "stale" by *t3* because a concurrent write might happen between *t2* and *t3*.
 
-etcd does not ensure linearizability for watch operations. Users are expected to verify the revision of watch responses to ensure correct ordering.
+etcd does not ensure linearizability for watch operations. The responses are ordered but may be stale. Users are expected to verify the revision of watch responses to ensure correct ordering against other operations.
```

Comment:
`ensure correct ordering` seems wrong. As we discussed, watch events are ordered, but their timing is unrelated to event execution. Revisions should be used when trying to figure out order of different operations.

Reply:
I have added `against other operations` at the end.  It may be just bad English on my part, I hoped to convey that it needs to be done ensure relative ordering of the watch responses and possible other responses.

As I understand it now, there can be a case where e.g. we send a range request for a key we are watching and we could theoretically first get revision r+1 from the range request and only then revision r from the watch request under certain scenarios. Or is that impossible?

How about `Users must compare revisions of watch responses against revisions in any other responses if relative order of these operations is needed.`?

Reply:
Proposed to change `against other operation` to `with other operations`. We are not comparing operations against each other, but ordering  with other operations. I think? I'm also not a native speaker :P